lib/internal/Magento/Framework/CompiledInterception/Generator/AreasPluginList.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\CompiledInterception\Generator;
+
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Config\Scope;
+
+/**
+ * Class AreasPluginList
+ */
+class AreasPluginList
+{
+    /** @var Scope */
+    private $scope;
+
+    /** @var array */
+    private $plugins;
+
```

Comment:
Why use private?  Your just destroying the ability to subclass properly.

Reply:
All properties by default should be private. The composition should be used instead of inheritance. Also, the list of plugins can be customized via a constructor.

Reply:
Both approaches has pros and cons but I guess consistency is most important and I assume "private by default " is M2 new standard.